src/clj_kondo/impl/linters.clj

```diff
@@ -102,12 +103,26 @@
                                (node->line (:filename ctx) expr :warning
                                            :constant-test-assertion "Test assertion with only constants.")))))
 
+(defn lint-single-key-in [ctx called-name call]
+  (let [keys (-> (:children call)
+                 (nth 2))
+        keys-count (-> keys
+                       (:children)
+                       (count))]
+    (when (and (= :vector (node/tag keys)) (= 1 keys-count))
```

Comment:
I think it's better to shortcut the linter in case the keys element is not a vector. So introduce a another when with this check and then continue.

Reply:
I don't quite understand. 
The docs say that `and` does shortcut so the only benefit of lifting the vector check out is that we wouldn't evaluate the `keys-count`. 

Is this a slow expression?